### PR TITLE
[PPANTT-120] feat: Implement Station Maintenances' APIs

### PR DESCRIPTION
**Pull request by svariant**
#### List of Changes

- Configured Feign Client
- Implemented client interface to [api-config](https://github.com/pagopa/pagopa-api-config) 
- Implemented, from [selfcare-backoffice](https://github.com/pagopa/pagopa-selfcare-ms-backoffice-backend), APIs to menage station maintenances
- Updated unit tests
- Updated openapis
- Excluded folder "util" from code review coverage

#### Motivation and Context

[PPANTT-120](https://pagopa.atlassian.net/jira/software/c/projects/PPANTT/boards/617?selectedIssue=PPANTT-120)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[PPANTT-120]: https://pagopa.atlassian.net/browse/PPANTT-120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "404", description = "Not Found",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
+            @ApiResponse(responseCode = "429", description = "Too many requests", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "500", description = "Service unavailable",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class)))
+    })
+    @OpenApiTableMetadata(readWriteIntense = OpenApiTableMetadata.ReadWrite.READ,
+            cacheable = true, external = true, internal = false)
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java:93]: why is cache yes?

File: src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    @GetMapping(value = "/{broker-tax-code}/summary", produces = {MediaType.APPLICATION_JSON_VALUE})
+    public MaintenanceHoursSummaryResource getBrokerMaintenancesSummary(
+            @Parameter(description = "Broker's tax code") @PathVariable("broker-tax-code") String brokerCode,
+            @Parameter(description = "Year of maintenance (yyyy)", example = "2024") @RequestParam @Size(min = 4, max = 4) String maintenanceYear
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java:97]: use `maintenance-year` as name in the swagger

File: src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            @ApiResponse(responseCode = "429", description = "Too many requests", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "500", description = "Service unavailable",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class)))
+    })
+    @OpenApiTableMetadata(readWriteIntense = OpenApiTableMetadata.ReadWrite.READ,
+            cacheable = true, external = true, internal = false)
+    @GetMapping(value = "/{broker-tax-code}", produces = {MediaType.APPLICATION_JSON_VALUE})
+    public StationMaintenanceListResource getStationMaintenances(
+            @Parameter(description = "Broker's tax code") @PathVariable("broker-tax-code") String brokerCode,
+            @Parameter(description = "Station's code") @RequestParam(required = false) String stationCode,
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/selfcare/pagopa/controller/StationMaintenanceController.java:55]: use `station-code` as name in the swagger

File: src/main/java/it/pagopa/selfcare/pagopa/model/stationmaintenance/MaintenanceHoursSummaryResource.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Model class that hold the hours' summary of stations' maintenance for a specific broker
+ */
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class MaintenanceHoursSummaryResource {
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/selfcare/pagopa/model/stationmaintenance/MaintenanceHoursSummaryResource.java:17]: specify the format for the fields of this class

**Review by jacopocarlini — CHANGES_REQUESTED** [review]
see the comments

File: src/main/java/it/pagopa/selfcare/pagopa/model/stationmaintenance/MaintenanceHoursSummaryResource.java (hunk 4 above)
Comment (svariant) [reply src/main/java/it/pagopa/selfcare/pagopa/model/stationmaintenance/MaintenanceHoursSummaryResource.java:17]: Hours are not in a specific date format, are composed of hours on base 10 and minutes on base 60. Example: 30 and half hours is "30:30", 30 and a quarter is "30:15", 30 is "30" without minutes

Comment (jacopocarlini) [reply src/main/java/it/pagopa/selfcare/pagopa/model/stationmaintenance/MaintenanceHoursSummaryResource.java:17]: Yes there isn't a date format but the client doesn't know how pass the value (ex: "30:15"). Can you put the example and a description in the swagger please?

**Comment by jacopocarlini** [comment]
resolve the new issue of sonar

**Pull request merged**